presto-main/src/test/java/com/facebook/presto/execution/controller/TestPidTaskExecutorController.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution.controller;
+
+import com.facebook.presto.execution.controller.PidTaskExecutorController.PidController;
+import io.airlift.units.Duration;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+import static org.testng.Assert.assertEquals;
+
+public class TestPidTaskExecutorController
+{
+    @Test
+    public void testLinearFunctions()
```

Comment:
OK I have no idea what's going on here, is there a link to some literature about PID you can put in the comments?